src/Compilers/RealParserTests/RandomRealParserTests.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```

Comment:
There are a countibly infinite number of possible strings that are valid floating-point literals; it is in principle *impossible* to test any nonzero fraction of them. Having said that, this approach did find bugs during development of this code, and it has been found to be a very effective way to find problems with this kind of code in general.